build-setup/vscode/using_vscode.md

```diff
@@ -2,45 +2,45 @@
 
 Previously, we explained how to [set up CLion for `ml-cpp`](../clion/using_clion.md). You may find, that CLion has some
 performance problems when indexing `ml-cpp` project: it may take very long to open the project or CLion may freeze and
-fail. In this case, [Visual Studio Code](https://code.visualstudio.com/) can be an alternative. When combined with
-`clangd`, it is significantly faster, while still being able to provide features like autocompletion, code navigation,
-static code analysis, etc.
+fail. In this case, [Visual Studio Code](https://code.visualstudio.com/) can be an alternative. `VSCode` offers support
+for `CMake` projects by way of an extension, while still being able to provide features like autocompletion,
+code navigation, static code analysis, etc. 
 
 ## Prerequisites
 
 In this tutorial, we assume that you have `gcc` or `clang` installed as described in the
-[CLion tutorial](../clion/using_clion.md). We also assume that you already created `compile_commands.json` file as
-described there since `clangd` also uses it.
-
-[Install `clangd`](https://clangd.llvm.org/installation.html). Version 11.0 had a significant performance improvement,
-so it is worth going the extra mile and install the latest stable version. Install
+[CLion tutorial](../clion/using_clion.md). 
+and
 [Visual Studio Code](https://code.visualstudio.com/) with
-[clangd plugin](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd).
+[cmake plugin](https://marketplace.visualstudio.com/items?itemName=twxs.cmake),
+[cmake tools plugin](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools),
+[C/C++ plugin](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools)
+as a minimum.
 
 In **Settings**, make sure *C_Cpp:Autocomplete* and *C_Cpp:Intelli Sense Engine* from the native C++ plugin of VS Code
 are disabled. Note that we don't want to disable the complete C/C++ Extension, since we are still going to use the
 debugging capabilities.
 
-Now for the `clangd` plugin, make sure that you have the correct path specified in **Settings** and *Clangd: Path*.
-*Clangd: Arguments* can look something like this:
+Now for the `CMake` plugin, change the `Build directory` specified in **Settings** > **Extensions** >
+**CMake Tools** to read `${workspaceFolder}/cmake_build`. You may also wish to set certain environment
+variables for the build. e.g. `ML_DEBUG=1`. To do so, under `Cmake: Build Environment` click the `Add item`
```

Comment:
Just seeing this here now makes me think we should change the `ML_DEBUG` functionality to instead be what you get when you do a debug build in CMake.  Then instead of setting environment variables a CLion user would just change the build type. Don't change it as part of this PR though.  We can come back to this at some time in the future.  It's not essential as hardly anyone does debug builds in reality, and it's also got dependencies on the Jenkins jobs, so needs some coordination there.